src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -538,21 +540,20 @@ public void StopStake()
                 else
                 {
                     this.logger.LogTrace("{0} failed to create POS block, waiting {1} ms for next round...", nameof(this.StakeAndSignBlockAsync), this.minerSleep);
-                    await Task.Delay(TimeSpan.FromMilliseconds(this.minerSleep), this.stakeCancellationTokenSource.Token).ConfigureAwait(false);
+                    await Task.Delay(TimeSpan.FromMilliseconds(this.minerSleep), cancellationToken).ConfigureAwait(false);
                 }
             }
         }
 
-        private async Task<List<UtxoStakeDescription>> GetUtxoStakeDescriptions(WalletSecret walletSecret, CancellationToken ct)
+        public async Task<List<UtxoStakeDescription>> GetUtxoStakeDescriptionsAsync(WalletSecret walletSecret, CancellationToken cancellationToken)
         {
             var utxoStakeDescriptions = new List<UtxoStakeDescription>();
-            IEnumerable<UnspentOutputReference> spendableTransactions = this.walletManager
+            IList<UnspentOutputReference> spendableTransactions = this.walletManager
                 .GetSpendableTransactionsInWallet(walletSecret.WalletName, 1)
                 .Where(t => t != null).ToList();
 
-            //todo 
-            FetchCoinsResponse fetchedCoinSet = await this.coinView.FetchCoinsAsync(spendableTransactions.Select(t => t.Transaction.Id).ToArray(), ct).ConfigureAwait(false);
-            if (ct.IsCancellationRequested) return utxoStakeDescriptions;
+            FetchCoinsResponse fetchedCoinSet = await this.coinView.FetchCoinsAsync(spendableTransactions.Select(t => t.Transaction.Id).ToArray(), cancellationToken).ConfigureAwait(false);
+            if (cancellationToken.IsCancellationRequested) return utxoStakeDescriptions;
```

Comment:
this should really NOT be needed

Reply:
In my opinion, if you want to cancel the task, then why would you bother to go through the other spendableTransactions in the loop ?

Reply:
You pass the token to FetchCoinsAsync, so if there is cancellation at any point before line 555, FetchCoinsAsync will throw exception. Therefore the chance of actually executing line 556 when the cancellation token is cancelled is terribly low. The line is not justified because it will be executed only in one of 10,000 attempts. And of course you're missing exit logs. But just delete it